slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -57,14 +57,43 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       else compoundValue(types)
     }
 
-    def compoundValue(values: Seq[String]): String = {
+    //build and nest tuples if > 22 elements
+    def tupleBuilder(values: Seq[String] , prefix: String): String = {
+      def mkTuple[T](l: Seq[T], _prefix: String) = l.map{ s => _prefix + s}.mkString("(" , " ," , ")")
+      def group[T](l: Seq[T] ) : Seq[ Seq[T] ] = {
+          if( l.isEmpty ) Seq[ Seq[T] ]()
+          else Seq[Seq[T]](l.take(22)) ++ group(l.drop(22) )
+      }
+      if(values.size <= 22) mkTuple(values,"") else mkTuple( group(values).map {mkTuple(_,prefix)} , "" )
```

Comment:
this breaks for tables > 484 columns, right? I have personally had tables > 1000 columns. Maybe one more nesting layer is enough? that would bring us to 10648.